addons/udes_stock/models/stock_picking_type.py

```diff
@@ -406,3 +433,100 @@ def get_picking_type(self, picking_type_id):
+        # u_num_reservable_pickings.
+        # However we must also take into account the atomic batch reservation
+        # flag (u_reserve_batches) and the handle partial flag
+        # (u_handle_partials).
+
+        for picking_type in picking_types:
+            _logger.info('Reserving stock for picking type %r.', picking_type)
+
+            # We want to reserve batches atomically, that is we will
+            # reserve pickings until all pickings in a batch have been
+            # assigned, even if we exceed the number of reservable pickings.
+            # However, the value of the handle partial flag is false we
+            # should not reserve stock if the batch cannot be completely
+            # reserved.
+            to_reserve = picking_type.u_num_reservable_pickings
+            reserve_all = picking_type.u_num_reservable_pickings == -1
+            base_domain = [
+                ('picking_type_id', '=', picking_type.id),
+                ('state', '=', 'confirmed')]
+            limit = 1
+
+            processed = StockPickings.browse()
+            while to_reserve > 0 or reserve_all:
+                domain = base_domain[:]
+                if processed:
+                    domain.append(('id', 'not in', processed.ids))
+                stock_pickings = pickings or StockPickings.search(domain,
+                                                                  limit=limit)
+
+                if not stock_pickings:
+                    # No pickings left to process.
+                    # If u_num_reservable_pickings is -1, or there are
+                    # fewer available pickings that the limit, the loop must
+                    # terminate here.
+                    break
+
+                batch = stock_pickings.mapped('batch_id')
+                if batch and batch.state == 'draft':
+                    # Add to seen pickings so that we don't try to process
+                    # this batch again.
+                    processed |= batch.picking_ids
+                    continue
+
+                if batch and picking_type.u_reserve_batches:
+                    stock_pickings = batch.picking_ids
+
+                # Assign at the move level because refactoring may change
+                # the pickings.
+                moves = stock_pickings.mapped('move_lines')
+                moves = moves._action_assign()
+                stock_pickings = moves.mapped('picking_ids')
+                processed |= stock_pickings
+
+                if not picking_type.u_handle_partials:
+                    if not all(x == 'assigned' for x in
+                               stock_pickings.mapped('state')):
+                        if pickings:
+                            # we need to construct our error message before the
+                            # changes are rolled back.
+                            key = lambda x: x.state != 'assigned'
+                            unsatisfied = stock_pickings.filtered(key)
```

Comment:
You can compute unsatisfied before line 511, and change condition of current line 511 to `if unsatisfied:`